org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/entitytemplates/artifacttemplates/ArtifactTemplateResource.java

```diff
@@ -86,6 +92,30 @@ public void synchronizeReferences() throws IOException {
 		BackendUtils.synchronizeReferences((ArtifactTemplateId) this.id);
 	}
 
+	@POST
+	@Consumes(MediaType.APPLICATION_JSON)
+	public Response copySourceToFilesResource(ArtifactResourcesApiData data) {
+		List<String> artifactList = data.getArtifactList();
+		ArtifactTemplateDirectoryId sourceDir = new ArtifactTemplateSourceDirectoryId((ArtifactTemplateId) this.id);
+		FilesResource filesResource = getFilesResource();
+		for (RepositoryFileReference ref : RepositoryFactory.getRepository().getContainedFiles(sourceDir)) {
+			if (artifactList == null || artifactList.contains(ref.getFileName())) {
+				try {
+					InputStream inputStream = RepositoryFactory.getRepository().newInputStream(ref);
+					String fileName = ref.getFileName();
+					String subDirectory = "";
```

Comment:
Does following Java8 thing work?

   subDirectory = ref.getSubDirectory().map(s -> s.toString()).orElse("");

This is shorter than the currently programmed four lines.